src/pages/ListPage.jsx

```diff
@@ -31,13 +37,18 @@ function ListPage() {
   ];
 
   const handleButtonClick = () => {
-    const data = localStorage.getItem("selectedSubject");
-    if (!data) {
+    const data = localStorage.getItem("subjects");
+    if (data) {
+      const subjects = JSON.parse(data);
+      if (subjects.length === 0) {
+        navigate("/");
+      } else {
+        setSubjects(subjects);
+        setIsModalOpen(true);
+      }
+    } else {
       navigate("/");
```

Comment:
```suggestion
    const data = localStorage.getItem("subjects");
    if (data) {
      const subjects = JSON.parse(data);
      if (subjects.length !== 0) {
        setSubjects(subjects);
        setIsModalOpen(true);
        return;
      }
    }
    navigate("/");
```
[제안] 아마 이런 식으로 작성하면 코드가 더 간결해지지 않을까 싶네요 (수정하실 거라면 아래 중괄호를 빼먹어서 직접 수정해주세요!)

Reply:
훨씬 간결하네요! 수정했습니다.